libraries/provider_pg_repluser.rb

```diff
@@ -0,0 +1,25 @@
+class Chef
+  class Provider
+    # adds replication user to database
+    class PgRepluser < Chef::Provider::LWRPBase
+      use_inline_resources if defined?(use_inline_resources)
+
+      def whyrun_supported?
+        true
+      end
+
+      action :create do
+        execute 'set-replication-user' do # ~FC009
+          role_exists = %(psql -c "SELECT rolname FROM pg_roles WHERE rolname='#{new_resource.repl_user}'" | grep #{new_resource.repl_user})
```

Comment:
if the pg gem would be useful anywhere else I would consdier adding it, but for this is not worth the effort and added dependencies